contract/src/lib.rs

```diff
@@ -85,119 +191,193 @@ impl Contract {
                     })
                 });
 
+        // Check if the content has already been purchased
+        assert!(
+            !content_purchases.contains(&id),
+            "Content already purchased"
+        );
+
+        let x = self.paid_contents.get(&id).unwrap();
+        let cost = x.cost;
+        let attached = env::attached_deposit().to_string();
+
+        // Check if the attached_deposit is wrong
+        assert!(
+            cost == attached,
+            "Attached deposit is wrong. It should be {} instead of {}",
+            cost,
+            attached
+        );
+
         // Add the content to the purchased list for the account
-        content_purchases.clear();
+        content_purchases.insert(&id);
         self.purchases_per_account
             .insert(&account_id, &content_purchases);
+
+        // Transfer attached deposit to the author
+        Promise::new(x.author).transfer(env::attached_deposit())
```

Comment:
The transfer will be performed in the next block. Can it fail? The content is purchased, but no tokens are transferred, for example.

Reply:
In the `test_buy_content` test I check if the number of buyer tokens decreases by the amount of the cost. I didn't find the ability to check if the author get these tokens 🤷